File: back/back/Models/Referral.cs
using System;
using System.Collections.Generic;

namespace back.Models;

public partial class Referral
{
    public long Id { get; set; }

    public ulong IsUsed { get; set; }

    public long? ForDoctorId { get; set; }

    public virtual ICollection<Appointment> Appointments { get; } = new List<Appointment>();

    public virtual User? ForDoctor { get; set; }
}

File: front/src/app/model/Referral.ts
import { Appointment } from './Appointement';
import { User } from './User';

export class Referral {
  id?: number = 0;
  isUsed: number = 0;
  forDoctorId: number = 0;
  appointments: Appointment[] = [];
  forDoctor: User | null = null;
}
